kbchat/kvstore_test.go

```diff
@@ -0,0 +1,60 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	res2, err2 := alice.PutEntry(team, namespace, key, "value1", 0)
```

Comment:
looking at this API, i'd say maybe we should be thinking about two methods for this instead of one: `PutEntry` and `PutEntryWithRevision`. this might play nicer with the types and branching logic above. and i'd argue it's not a problem from a usability perspective to have extra methods for Put and Delete that handle this complexity.

Reply:
also, `err2`?
i suggest adding a `var err error` up around line 18 and then just use `err` for all your errors.